here-naksha-lib-core/src/main/java/com/here/naksha/lib/core/AbstractTask.java

```diff
@@ -581,4 +597,83 @@ public final boolean removeListener(@NotNull Consumer<@NotNull RESULT> listener)
       state.set(State.NEW);
     }
   }
+
+  /**
+   * Increments the value of author usage for given actor anc compares with the specified limit.
+   *
+   * <p>This method ensures that the number of concurrent tasks for the actor
+   * does not exceed the specified limit. If the limit is reached, it logs an
+   * error and throws a {@link TooManyTasks} exception.
+   *
+   * @param actorId The identifier of the actor for which to acquire the slot.
+   * @param limit The maximum number of concurrent tasks allowed for the actor.
+   * @throws TooManyTasks If the maximum number of concurrent tasks is reached for the actor.
+   */
+  private void incActorLevelUsage(String actorId, long limit) {
+    if (internal || actorId == null) return;
```

Comment:
Internal tasks are created by Naksha right? They are not related to any app or author? Also for internal tasks actor id is null? How can we handle increment and decrement?

Reply:
Internal as well always have a context and must have an app_id, optionally with a actor. Internal tasks happen for certain reasons, but most of the time they are forked as children of a main task. We do count them the same way we count all other tasks, but they ignore limits. The reason is that we do not want to break code internally. Assume the client asks to query a view and he has one slot left, the thread is created and processsing starts. Then at the end of the pipeline, there is a view which want to start 4 sub-queries, which means, start 4 child tasks. These children will be counted against the limits, but they will not be prevented, otherwise we would break the initial approval of the request. Therefore, limits are soft-caps, we just prevent that a customer initially start more than X tasks, but overall he may use more tasks, like 10 or 20 more, when the requests require sub-tasks. Does this clarify the internal?

Reply:
Got it. Thanks for the clarification. Removing checks for internal in both incActorLevelUsage & decActorLevelUsage.